examples/md-flexible/src/domainDecomposition/RegularGridDecomposition.cpp

```diff
@@ -173,49 +185,55 @@ void RegularGridDecomposition::exchangeHaloParticles(AutoPasType &autoPasContain
       continue;
     }
 
-    auto particlesForLeftNeighbor = collectHaloParticlesForLeftNeighbor(autoPasContainer, dimensionIndex);
-    auto particlesForRightNeighbor = collectHaloParticlesForRightNeighbor(autoPasContainer, dimensionIndex);
+    _particlesForLeftNeighbor.clear();
+    collectHaloParticlesForLeftNeighbor(autoPasContainer, dimensionIndex, _particlesForLeftNeighbor);
+    _particlesForRightNeighbor.clear();
+    collectHaloParticlesForRightNeighbor(autoPasContainer, dimensionIndex, _particlesForRightNeighbor);
 
-    const double leftHaloMin = _localBoxMin[dimensionIndex] - _skinWidthPerTimestep * _rebuildFrequency;
-    const double leftHaloMax = _localBoxMin[dimensionIndex] + _cutoffWidth + _skinWidthPerTimestep * _rebuildFrequency;
-    const double rightHaloMin = _localBoxMax[dimensionIndex] - _cutoffWidth - _skinWidthPerTimestep * _rebuildFrequency;
-    const double rightHaloMax = _localBoxMax[dimensionIndex] + _skinWidthPerTimestep * _rebuildFrequency;
+    const double leftHaloMin = _localBoxMin[dimensionIndex] - skinWidth;
+    const double leftHaloMax = _localBoxMin[dimensionIndex] + _cutoffWidth + skinWidth;
+    const double rightHaloMin = _localBoxMax[dimensionIndex] - _cutoffWidth - skinWidth;
+    const double rightHaloMax = _localBoxMax[dimensionIndex] + skinWidth;
 
-    for (const auto &particle : haloParticles) {
+    for (const auto &particle : _haloParticles) {
```

Comment:
Not sure if I'm blind.
`_haloParticles' is cleared in l.167, but not filled again until here? So I'm not sure what is supposed to happen in this loop

Reply:
`_haloParticles` is filled in line 227. Thus, starting from the second iteration of the outer loop (line 179) `_haloParticles` contains particles and we enter the loop.

Reply:
Ah okay, I guess then this loop is there to send particles across corners?